apps/meteor/ee/app/livechat-enterprise/server/api/lib/priorities.ts

```diff
@@ -53,13 +53,29 @@ export async function updatePriority(
 		}),
 	};
 
-	const created = await LivechatPriority.findOneAndUpdate(query, update, {
-		returnDocument: 'after',
-	});
+	try {
+		if (data.name) {
+			const priority = await LivechatPriority.findOne({ name: new RegExp(`^${escapeRegExp(data.name)}$`, 'i') });
+			if (priority && priority._id !== _id) {
+				throw new Error('Error! Priority already exists with this name');
+			}
+		}
 
-	if (!created.ok || !created.value) {
-		throw Error('Error updating priority');
-	}
+		const created = await LivechatPriority.findOneAndUpdate(query, update, {
+			returnDocument: 'after',
+		});
+
+		if (!created.ok || !created.value) {
+			throw Error('Error updating priority');
+		}
 
-	return created.value;
+		return created.value;
+	} catch (error: any) {
+		// check if its a duplicate key error
+		if (error?.code === 11000) {
+			throw new Error('Error! Priority already exists with this name');
```

Comment:
Hey Murtaza. I was talking with @KevLehman and it might be a good idea to change this message to a consistent error code (Ex.: error-duplicated-name). That way is easier to check and translate the message on the frontend.

Reply:
Makes sense :+1: Done with the change, please check :wink: